misc/scripts/library-coverage/compare-files-comment-pr.py

```diff
@@ -0,0 +1,183 @@
+            folder1 + "/" + generated_output_csv, folder2 + "/" + generated_output_csv)
+
+        if cmp1 != "" or cmp2 != "":
+            print("Generated file contents are not matching", file=sys.stderr)
+            return_md += "\n### " + lang + "\n\n#### Generated file changes for " + \
+                lang + "\n\n"
+            if cmp1 != "":
+                return_md += "- Changes to " + generated_output_rst + \
+                    ":\n```diff\n" + cmp1 + "```\n\n"
+            if cmp2 != "":
+                return_md += "- Changes to " + generated_output_csv + \
+                    ":\n```diff\n" + cmp2 + "```\n\n"
+
+    with open(output_file, 'w', newline='') as out:
+        out.write(return_md)
+
+
+comment_pr(sys.argv[1], sys.argv[2], sys.argv[3],
+           sys.argv[4], sys.argv[5], sys.argv[6])
+
+
+# def compare_generated_and_repo_files():
+#     languages = ['java']
+
+#     all_ok = True
+
+#     for lang in languages:
+#         repo_output_rst = settings.repo_output_rst.format(language=lang)
+#         repo_output_csv = settings.repo_output_csv.format(language=lang)
+
+#         generated_output_rst = settings.generated_output_rst.format(
+#             language=lang)
+#         generated_output_csv = settings.generated_output_csv.format(
+#             language=lang)
+
+#         exists = check_file_exists(repo_output_rst)
+#         if not exists:
+#             sys.exit(1)
+
+#         exists = check_file_exists(repo_output_csv)
+#         if not exists:
+#             sys.exit(1)
+
+#         exists = check_file_exists(generated_output_rst)
+#         if not exists:
+#             sys.exit(1)
+
+#         exists = check_file_exists(generated_output_csv)
+#         if not exists:
+#             sys.exit(1)
+
+#         docs_folder = settings.documentation_folder_no_prefix.format(
+#             language=lang)
+
+#         rst_ok = compare_files(repo_output_rst, generated_output_rst)
+#         if not rst_ok:
+#             print("The generated file doesn't match the one in the codebase. Please check and fix file '" +
+#                   docs_folder + settings.output_rst_file_name + "'.", file=sys.stderr)
+#         csv_ok = compare_files(repo_output_csv, generated_output_csv)
+#         if not csv_ok:
+#             print("The generated file doesn't match the one in the codebase. Please check and fix file '" +
+#                   docs_folder + settings.output_csv_file_name + "'.", file=sys.stderr)
+
+#         if not rst_ok or not csv_ok:
+#             print("The generated CSV coverage report files for '" + lang + "' don't match the ones in the codebase. Please update the files in '" +
+#                   docs_folder + "'. The new files can be downloaded from the artifacts of this job.", file=sys.stderr)
+#             all_ok = False
+#         else:
+#             print("The generated files for '" + lang +
+#                   "' match the ones in the codebase.")
+
+#     if not all_ok:
+#         sys.exit(1)
```

Comment:
There is lots of commented-out code above this line. Should that be removed?

Reply:
Yes, I'll remove these, and some of it will come back as part of the followup work on the automatic PR creation with the changes in `main`.